appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/interceptors/RetryInterceptor.java

```diff
@@ -233,29 +233,29 @@ private Object retry(InvocationContext invocationContext) throws Exception {
                 throw ex;
             }
 
-            int maxRetries = (Integer) FaultToleranceCdiUtils.getOverrideValue(
+            int maxRetries = FaultToleranceCdiUtils.getOverrideValue(
                 config, Retry.class, "maxRetries", invocationContext, Integer.class)
                 .orElse(retry.maxRetries());
-            long delay = (Long) FaultToleranceCdiUtils.getOverrideValue(
+            long delay = FaultToleranceCdiUtils.getOverrideValue(
                 config, Retry.class, "delay", invocationContext, Long.class)
                 .orElse(retry.delay());
             // Look for a String and cast to ChronoUnit - Use the Common Sense Convertor
-            ChronoUnit delayUnit = (ChronoUnit) FaultToleranceCdiUtils.getOverrideValue(
-                config, Retry.class, "delayUnit", invocationContext, String.class)
+            ChronoUnit delayUnit = FaultToleranceCdiUtils.getOverrideValue(
+                config, Retry.class, "delayUnit", invocationContext, String.class).map(ChronoUnit::valueOf)
```

Comment:
Note the `map`. I assume that when loaded from config the `ChronoUnit` is given as `String` and has to be mapped to a `ChronoUnit`. Same two times more below.

Reply:
No I think the original code was simply wrong actually. Should have been:
`ChronoUnit delayUnit = (ChronoUnit) FaultToleranceCdiUtils.getOverrideValue(config, Retry.class, "delayUnit", invocationContext, **ChronoUnit**.class)`

See the this commit that fixes this elsewhere: https://github.com/payara/Payara/commit/8044cf2a5d7265dcb5b2137540a7eb7e41772af6#diff-709d4edc61c9c09ffdaff2ba0f9bc4a7

Reply:
Okay, this was actually my first assumption but when I found the case where `Class` was extracted I started to think that the `Config` actually contains the value as `String`. Will change these.